### PR TITLE
[#37] Set up new Next.JS 13 application

### LINKED ISSUE #37: Set up new Next.JS 13 application
## Why

Before adding features, the framework must be set up with the provided generator. It allows to check the various options available when generating a new project and assess the support for Web and API variants.

## Acceptance Criteria

- Create a new directory in the project's root with the framework's name.
- Generate a barebone project using the provided generator.
- Ensure the application can be started simultaneously with the other framework. So a different local port for each application must be used.

### PR DESCRIPTION
**Pull request by tednguyendev**
Close #37

## What happened 👀
- Use the Next.JS CLI with the typescript option
- Add the basic folders structure

## Insight 📝
Created the application using `create-next-app`.

## Proof Of Work 📹
CleanShot 2023-05-19 at 17 32 56

### CODE REVIEW

**Comment by phisith** [comment]
Good morning Ted, For the folder structure I think you are missing the api directory under the pages directory.

As we know that in NextJS, it can act itself as the nodeJs server, like you can create api endpoint in your project, you can read more about it from the link below : )

 https://nextjs.org/docs/pages/building-your-application/routing/api-routes
 https://github.com/vercel/next.js/tree/canary/examples/api-routes-rest

**Comment by tednguyendev** [comment]
> Good morning Ted, For the folder structure I think you are missing the api directory under the pages directory.
> 
> As we know that in NextJS, it can act itself as a nodeJs server, like you can create api endpoint in your project, you can read more about it from the link below : )
> 
> https://nextjs.org/docs/pages/building-your-application/routing/api-routes https://github.com/vercel/next.js/tree/canary/examples/api-routes-rest

Oh I see, ok [added](https://github.com/nimblehq/js-framework-benchmark/pull/76/commits/238c4f867333f0495c9ca92cfeb127c3f48fcc94)! Thank you :D Beckham

**Comment by phisith** [comment]
For the rest, I think is looking great! 
Not sure that we need Olivier to double it before Merge 😅

File: nextjs-13/README.md
Comment (olivierobert) [inline nextjs-13/README.md:1]: You need to customize this README to fit each sub-application (you should not reuse the one in the root directory). The README must contain information on hos to runt the application. You check the one for the Next.JS 12 application: https://github.com/nimblehq/js-framework-benchmark/blob/develop/nextjs-12/README.md

**Comment by tednguyendev** [comment]
Yeah right 🤔 Then I think I will just leave this PR here for a while, it's not urgent so it's okay :D

File: nextjs-13/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "nextjs-13",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev -p 3003",
```
Comment (olivierobert) [inline nextjs-13/package.json:6]: Minor preference to change the port at the hundredth unit:

```suggestion
    "dev": "next dev -p 3300",
```

It allows to use sub-ports `3301`, `3302`, etc. for the same application when needed.

**Review by olivierobert — CHANGES_REQUESTED** [review]

File: nextjs-13/package.json (hunk 1 above)
Comment (tednguyendev) [reply nextjs-13/package.json]: Woa TIL! [This is the commit with the change](https://github.com/nimblehq/js-framework-benchmark/pull/76/commits/26865c6516980236ce256455a727757d85762092).

File: nextjs-13/README.md
Comment (tednguyendev) [reply nextjs-13/README.md:1]: And this is my commit to update the README to be suitable for this sub-application: https://github.com/nimblehq/js-framework-benchmark/pull/76/commits/acaa253ffa907a68f55342948f5e7ff1bfa8c9f2

**Review by olivierobert — APPROVED** [review]

**Review by phisith — APPROVED** [review]

**Pull request merged**